From 432cf7600847c1a8f7b679a7efec4035ec09f2ae Mon Sep 17 00:00:00 2001
From: Doug Ledford <dledford@redhat.com>
Date: Wed, 24 Sep 2014 15:22:07 -0400
Subject: [infiniband] ocrdma: Return proper value for max_mr_size

Message-id: <653f121608b7c42ee66fbe2603eef51438eab63b.1411571481.git.dledford@redhat.com>
Patchwork-id: 96639
O-Subject: [Patch RHEL7 32/48] RDMA/ocrdma: Return proper value for max_mr_size
Bugzilla: 1081067 1088577
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Mitesh Ahuja <mitesh.ahuja@emulex.Com>

Update the max_mr_size with proper value. Corrected the response
structure of query config mailbox command.

Signed-off-by: Mitesh Ahuja <mitesh.ahuja@emulex.Com>
Signed-off-by: Selvin Xavier <selvin.xavier@emulex.com>
Signed-off-by: Roland Dreier <roland@purestorage.com>
(cherry picked from commit 033edd4dff23708c35120d9297fc898ef284bc7d)
Signed-off-by: Doug Ledford <dledford@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/infiniband/hw/ocrdma/ocrdma_hw.c b/drivers/infiniband/hw/ocrdma/ocrdma_hw.c
index 5b6e9d9..105659d 100644
--- a/drivers/infiniband/hw/ocrdma/ocrdma_hw.c
+++ b/drivers/infiniband/hw/ocrdma/ocrdma_hw.c
@@ -1088,7 +1088,8 @@ static void ocrdma_get_attr(struct ocrdma_dev *dev,
      OCRDMA_MBX_QUERY_CFG_CA_ACK_DELAY_SHIFT;
  attr->max_mw = rsp->max_mw;
  attr->max_mr = rsp->max_mr;
- attr->max_mr_size = ~0ull;
+ attr->max_mr_size = ((u64)rsp->max_mr_size_hi << 32) |
+         rsp->max_mr_size_lo;
  attr->max_fmr = 0;
  attr->max_pages_per_frmr = rsp->max_pages_per_frmr;
  attr->max_num_mr_pbl = rsp->max_num_mr_pbl;
diff --git a/drivers/infiniband/hw/ocrdma/ocrdma_sli.h b/drivers/infiniband/hw/ocrdma/ocrdma_sli.h
index 14a84b2..3cb88f0 100644
--- a/drivers/infiniband/hw/ocrdma/ocrdma_sli.h
+++ b/drivers/infiniband/hw/ocrdma/ocrdma_sli.h
@@ -530,8 +530,8 @@ struct ocrdma_mbx_query_config {
  u32 max_ird_ord_per_qp;
  u32 max_shared_ird_ord;
  u32 max_mr;
- u32 max_mr_size_lo;
  u32 max_mr_size_hi;
+ u32 max_mr_size_lo;
  u32 max_num_mr_pbl;
  u32 max_mw;
  u32 max_fmr;
diff --git a/drivers/infiniband/hw/ocrdma/ocrdma_verbs.c b/drivers/infiniband/hw/ocrdma/ocrdma_verbs.c
index 8cd16a1..90152de 100644
--- a/drivers/infiniband/hw/ocrdma/ocrdma_verbs.c
+++ b/drivers/infiniband/hw/ocrdma/ocrdma_verbs.c
@@ -69,7 +69,7 @@ int ocrdma_query_device(struct ib_device *ibdev, struct ib_device_attr *attr)
  memcpy(&attr->fw_ver, &dev->attr.fw_ver[0],
         min(sizeof(dev->attr.fw_ver), sizeof(attr->fw_ver)));
  ocrdma_get_guid(dev, (u8 *)&attr->sys_image_guid);
- attr->max_mr_size = ~0ull;
+ attr->max_mr_size = dev->attr.max_mr_size;
  attr->page_size_cap = 0xffff000;
  attr->vendor_id = dev->nic_info.pdev->vendor;
  attr->vendor_part_id = dev->nic_info.pdev->device;
-- 
1.7.1